### PR TITLE
Eth2 slashing protection API

### PR DESCRIPTION
**Pull request by jframe**
Adds additional fields to the Eth2 signing request to support slashing

#192

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/service/JacksonSerializers.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  public static class Base64Serialiser extends JsonSerializer<Bytes> {
+    @Override
+    public void serialize(
+        final Bytes value, final JsonGenerator gen, final SerializerProvider serializers)
+        throws IOException {
+      gen.writeString(value.toBase64String());
+    }
+  }
+
+  public static class HexDeserialiser extends JsonDeserializer<Bytes> {
```
Comment (usmansaleem) [inline core/src/main/java/tech/pegasys/web3signer/core/service/JacksonSerializers.java:52]: won't it conflict with Base64Deserialiser? both of them are converting Bytes?

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/SigningJsonRpcModule.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+
+public class SigningJsonRpcModule extends SimpleModule {
+
+  public SigningJsonRpcModule() {
+    super("SigningJsonRpcModule");
+    addDeserializer(Bytes.class, new HexDeserialiser());
+    addSerializer(Bytes.class, new HexSerialiser());
+    addDeserializer(UInt64.class, new StringUInt64Deserializer());
```
Comment (usmansaleem) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/SigningJsonRpcModule.java:30]: oh got it. Only FC will use Base64Deserializer.

File: core/src/main/java/tech/pegasys/web3signer/core/service/JacksonSerializers.java (hunk 1 above)
Comment (jframe) [reply core/src/main/java/tech/pegasys/web3signer/core/service/JacksonSerializers.java:52]: Only if you use both at the same time. This class doesn't register any of the deserializers it's merely a factory, that is up to the Jackson module.

**Review by usmansaleem — APPROVED** [review]
Looks good to me.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/SignerConfiguration.java [hunk 3]
```diff
@@ -32,6 +32,7 @@
   private Optional<TlsOptions> serverTlsOptions;
   private Optional<TlsCertificateDefinition> overriddenCaTrustStore;
   private final int metricsPort;
+  private final boolean slashingProtectionEnabled;
```
Comment (rain-on) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/SignerConfiguration.java:35]: you can probably just put in a URL - if its not null, then slashing is enabled, and you can set the url at the same time - I think I have this in one of my PRs?

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/runner/Web3SignerRunner.java [hunk 4]
```diff
@@ -120,6 +120,8 @@ public void shutdown() {
       params.add("--data-path");
       params.add(dataPath.toAbsolutePath().toString());
     }
+    params.add("--slashing-protection-enabled");
```
Comment (rain-on) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/runner/Web3SignerRunner.java:123]: again - take a look at one of my PRs, think this is done with a URL

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/SignerConfiguration.java (hunk 3 above)
Comment (jframe) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/SignerConfiguration.java:35]: removing it, don't need to disable slashing protection as still using noop slashing protection. will just use the url when we merge in real slashing protection

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/runner/Web3SignerRunner.java (hunk 4 above)
Comment (jframe) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/runner/Web3SignerRunner.java:123]: removed it

**Pull request merged**